backend/course/crawler.py

```diff
@@ -0,0 +1,121 @@
+import requests
+from bs4 import BeautifulSoup as bs
+import xlrd
+
+
+class Crawler:
+    search_url = 'http://sugang.snu.ac.kr/sugang/cc/cc100.action'
+    excel_url = 'http://sugang.snu.ac.kr/sugang/cc/cc100excel.action'
+    semester_name = {
+        'U000200001U000300001': '1',
+        'U000200001U000300002': 'S',
+        'U000200002U000300001': '2',
+        'U000200002U000300002': 'W'
+    }
+    # srchOpenSchyy : year
+    # srchOpenShtm : semester
+    # srchOpenUpSbjtFldCd : area
+    # srchOpenSbjtFldCd : subarea
+    form_data = {
+        'srchOpenUpSbjtFldCd': '',
+        'srchOpenSbjtFldCd': '',
+        'srchOpenShtm': '',
+        'srchCond': '1',
+        'srchOpenSchyy': '',
+        'srchCptnCorsFg': '',
+        'srchOpenShyr': '',
+        'srchSbjtCd': '',
+        'srchSbjtNm': '',
+        'srchOpenUpDeptCd': '',
+        'srchOpenDeptCd': '',
+        'srchOpenMjCd': '',
+        'srchOpenSubmattCorsFg': '',
+        'srchOpenSubmattFgCd': '',
+        'srchOpenPntMin': '',
+        'srchOpenPntMax': '',
+        'srchCamp': '',
+        'srchBdNo': '',
+        'srchProfNm': '',
+        'srchTlsnAplyCapaCntMin': '',
+        'srchTlsnAplyCapaCntMax': '',
+        'srchTlsnRcntMin': '',
+        'srchTlsnRcntMax': '',
+        'srchOpenSbjtTmNm': '',
+        'srchOpenSbjtTm': '',
+        'srchOpenSbjtTmVal': '',
+        'srchLsnProgType': '',
+        'srchMrksGvMthd': '',
+        'srchFlag': '',
+    }
+
+    def parse_option_tag(self, op):
+            return {'code': op['value'], 'name': op.text.strip()}
+
+    def get_areas(self, year, semester):
```

Comment:
It seems that `get_areas()`, `get_subareas()`, `get_courses()` functions have quite redundant code.
Would there be any way to make it more simple?